src/PersistStore.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  action,
+  IReactionDisposer,
+  isAction,
+  isComputedProp,
+  makeObservable,
+  observable,
+  reaction,
+  runInAction,
+  toJS,
+} from 'mobx';
+import { StorageConfiguration } from './StorageConfiguration';
+import { PersistenceOptions } from './types';
+
+export class PersistStore<T> {
+  private cancelWatch: IReactionDisposer | null = null;
+  private options: PersistenceOptions | null = null;
+  private target: T | null = null;
+
+  public isHydrated = false;
+  public isPersisting = false;
+
+  constructor(options: PersistenceOptions, target: T) {
+    this.options = options;
+    this.target = target;
+
+    makeObservable(
+      this,
+      {
+        clearPersist: action,
+        disposePersist: action,
+        isHydrated: observable,
+        isPersisting: observable,
+        rehydrateStore: action,
+        startPersist: action,
+        stopPersist: action,
+      },
+      { autoBind: true, deep: false },
+    );
+
+    this.init();
+  }
+
+  private async init() {
+    await this.rehydrateStore();
+    this.startPersist();
+  }
+
+  public async rehydrateStore(): Promise<void> {
```

Comment:
This public method should take care of https://github.com/quarrant/mobx-persist-store/issues/23 ?